src/lib/translators/mongo.ts

```diff
@@ -432,7 +432,79 @@ function transformPivot(step: Readonly<S.PivotStep>): MongoStep[] {
   ];
 }
 
-/** transform an 'replace' step into corresponding mongo steps */
+/** transform a 'rank' step into corresponding mongo steps */
+function transformRank(step: Readonly<S.RankStep>): MongoStep {
+  if (step.column) {
+    return [
```

Comment:
I think this solution is fine for dense ranking ([10, 15, 15, 20, 30] => [1, 2, 2, 3, 4]).
But in lots of cases it's not the expected ranking method, and you want to "skip" the rank of elements following duplicates ([10, 15, 15, 20, 30] => [1, 2, 2, 4, 5]) so that at the end the rank remains in line with the cumulated count of elements.
So we should make this an option in the step maybe (`dense: <true/false>`, `false` by default)

A nice inspiration of how to implement this: http://www.kamsky.org/stupid-tricks-with-mongodb/rank-and-dense-rank-in-aggregation